src/components/OverlayImage.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect, useReducer } from "react"
+import { styled } from "@storybook/theming";
+
+const movementReducer = (state, offset) => {
+  return {
+    x: state.x + offset.x,
+    y: state.y + offset.y,
+  }
+}
+
+interface OverlayImageProps {
+  url: string;
+  opacity: number;
+  scaling: number;
+  isLocked: boolean;
+  showDifference: boolean;
+}
+
+const OverlayImage = ({ url, opacity, scaling, isLocked, showDifference }: OverlayImageProps) => {
+  const [position, updatePosition] = useReducer(movementReducer, {x: 0, y: 0});
```

Comment:
I should've added eslint config for this though.
```suggestion
  const [position, updatePosition] = useReducer(movementReducer, { x: 0, y: 0 });
```